File: server/cinemadle_datamodel/media.go
package cinemadle_datamodel

import "encoding/json"

type Media struct {
	Id      int `json:"id"`
	Length  int `json:"length"`
	Revenue int `json:"revenue"`

	VoteAverage float32 `json:"revenue"`

	Title     string `json:"title"`
	Rating    string `json:"rating"`
	Date      string `json:"date"`
	PosterUri string `json:posterUri"`

	Cast []Person `json:"cast"`
	Crew []Person `json:"crew"`
}

func (m Media) ToJson() (string, error) {
	j, err := json.Marshal(m)

	if err != nil {
		return "", err
	}

	return string(j), nil
}

File: backend/CinemadleCore/DataModel/Media.cs
namespace CinemadleCore.DataModel;

public readonly struct Media
{
    public readonly int Id;

    public readonly int Length;

    public readonly int Revenue;

    public readonly double VoteAverage;

    public readonly string Title;

    public readonly string Rating;

    public readonly string Date;

    public readonly string ImageUri;

    public readonly IEnumerable<Person> Cast;

    public readonly IEnumerable<Person> Crew;
}